src/portfft/committed_descriptor_impl.hpp

```diff
@@ -203,12 +215,12 @@ class committed_descriptor_impl {
    * set of kernels that need to be JIT compiled.
    *
    * @tparam SubgroupSize size of the subgroup
-   * @param kernel_num the consecutive number of the kernel to prepare
+   * @param fft_size The size of the dft transform
    * @return implementation to use for the dimension and a vector of tuples of: implementation to use for a kernel,
    * vector of kernel ids, factors
    */
   template <Idx SubgroupSize>
-  std::tuple<detail::level, kernel_ids_and_metadata_t> prepare_implementation(std::size_t kernel_num) {
+  std::tuple<detail::level, std::size_t, kernel_ids_and_metadata_t> prepare_implementation(IdxGlobal fft_size) {
```

Comment:
Its to avoid an additional casts inside prepare implementation, which currently works with IdxGlobal,
note that even previously we used to cast the size to IdxGlobal inside prepare implementation, instead this directly accepts an IdxGlobal

It returns a size_t because a lot of our calculation pertaining to that field works on size_t, hence returning size_t to avoid casts there

Reply:
We use std::size_t in public interface and IdxGlobal in kernels. Somewhere inbetween we need to cast from one type to another. I just think this should happen just once, without any casts back to std::size_t. I am not particularly attached to where that is.

Reply:
The field `length` in the dimension_struct is stored as std::size_t, which is passed to other functions like num_scalars_in_local_mem, which accepts it as a size_t, hence that cast to size_t upon return.

we used to cast the fft_size to IdxGlobal inside the prepare_implementation [here](https://github.com/codeplaysoftware/portFFT/blob/653fde89ca16f3b08a925c8a06d1f1a391e90f21/src/portfft/committed_descriptor_impl.hpp#L220) previously presumably to avoid warnings, and hence seeing that I simply made it accept IdxGlobal

Reply:
But that requires additional casts, which make code harder to read. Make it so that we only cast to IdxGlobal, not back.